analyzers/src/SonarAnalyzer.VisualBasic/Rules/ExtensionMethodShouldNotExtendObject.cs

```diff
@@ -0,0 +1,47 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.VisualBasic;
+
+[DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+public sealed class ExtensionMethodShouldNotExtendObject : ExtensionMethodShouldNotExtendObjectBase<SyntaxKind>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+    protected override bool IsExtensionMethod(SyntaxNode methodDeclaration) =>
+        methodDeclaration.Parent.Parent is ModuleBlockSyntax
+        && ((MethodStatementSyntax)methodDeclaration).AttributeLists
+            .SelectMany(x => x.Attributes)
+            .Any(IsExtension);
+
+    private bool IsExtension(AttributeSyntax attribute)
```

Comment:
This is quite repetitive problem. On C# side, we have
https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.CSharp/Extensions/AttributeSyntaxExtensions.cs

We could copy/paste it to VB project as well for symetry.

Reply:
What I meant was to have `AttributeSyntaxExtensions.IsKnownType` on VB side, just as we have it on C#. And use that. The `IsExtension` is too specific to be an extension by itself.